tests/flask/test_blueprint.py

```diff
@@ -12,167 +15,27 @@ def client():
-    response = client.get("/vulnpy/deserialization")
-    assert response.status_code == 200
-
-
-def test_cmdi_os_system_get(client):
-    response = client.get("/vulnpy/cmdi/os-system/?user_input=echo%20attack")
-    assert response.status_code == 200
-
-
-def test_cmdi_os_system_post(client):
-    response = client.post(
-        "/vulnpy/cmdi/os-system/", data={"user_input": "echo attack"}
-    )
-    assert response.status_code == 200
-
-
-def test_cmdi_os_system_bad_command(client):
-    response = client.get("/vulnpy/cmdi/os-system/?user_input=foo")
-    assert response.status_code == 200
-
-
-def test_cmdi_os_system_invalid_input(client):
-    response = client.get("/vulnpy/cmdi/os-system/?ignored_param=bad")
-    assert response.status_code == 200
-
-
-def test_cmdi_subprocess_popen_get(client):
-    response = client.get("/vulnpy/cmdi/subprocess-popen/?user_input=echo%20attack")
-    assert response.status_code == 200
-
-
-def test_cmdi_subprocess_popen_post(client):
-    response = client.post(
-        "/vulnpy/cmdi/subprocess-popen/", data={"user_input": "echo attack"}
-    )
-    assert response.status_code == 200
-
-
-def test_cmdi_subprocess_popen_bad_command(client):
-    response = client.get("/vulnpy/cmdi/subprocess-popen/?user_input=foo")
-    assert response.status_code == 200
-
-
-def test_cmdi_subprocess_popen_invalid_input(client):
-    response = client.get("/vulnpy/cmdi/os-system/?ignored_param=bad")
-    assert response.status_code == 200
-
-
-def test_deserialization_pickle_load_get(client):
-    response = client.get(
-        "/vulnpy/deserialization/pickle-load/?user_input={}".format(
-            "csubprocess\ncheck_output\n(S'ls'\ntR."
-        )
-    )
-    assert response.status_code == 200
-
-
-def test_deserialization_pickle_load_post(client):
-    response = client.post(
-        "/vulnpy/deserialization/pickle-load/",
-        data={"user_input": "csubprocess\ncheck_output\n(" "S'ls'\ntR."},
-    )
+@parametrize_root
+def test_root_views(client, view_name):
+    if view_name == "home":
+        response = client.get("/vulnpy")
+    else:
+        response = client.get("/vulnpy/{}".format(view_name))
     assert response.status_code == 200
 
 
-def test_deserialization_pickle_loads_get(client):
-    response = client.get(
-        "/vulnpy/deserialization/pickle-loads/?user_input={}".format(
-            "csubprocess\ncheck_output\n(S'ls'\ntR."
-        )
-    )
-    assert response.status_code == 200
-
+@parametrize_triggers
+@pytest.mark.parametrize("request_method", ["get", "post"])
+def test_trigger(client, request_method, view_name, trigger_name):
+    get_or_post = getattr(client, request_method)
 
-def test_deserialization_pickle_loads_post(client):
-    response = client.post(
-        "/vulnpy/deserialization/pickle-loads/",
-        data={"user_input": "csubprocess\ncheck_output\n(S'ls'\ntR."},
-    )
-    assert response.status_code == 200
+    data = DATA.get(view_name)
```

Comment:
instead of using `.get()`, can we do an unsafe lookup? Since this is test code, I think it makes sense for it to fail catastrophically if there's no corresponding attack vector

Reply:
I like it!